Src/Amr/AMReX_extrapolater_2D_K.H

```diff
@@ -0,0 +1,143 @@
+#ifndef AMReX_extrapolater_2D_K_H_
+#define AMReX_extrapolater_2D_K_H_
+
+namespace amrex {
+
+AMREX_GPU_HOST_DEVICE
+AMREX_FORCE_INLINE
+void
+amrex_first_order_extrap(amrex::Box const& bx,
+                         int               nComp,
+                         amrex::Array4<const int>   const& mask,
+                         amrex::Array4<amrex::Real> const& data) noexcept
+{
+   using namespace amrex::literals;
+
+   const int finecell = 1;
+   const int crsecell = 0;
+
+   const auto lo = amrex::lbound(bx);
+   const auto hi = amrex::ubound(bx);
+   int idlo[3] = {AMREX_D_DECL(lo.x,lo.y,lo.z)};            // base box bounds
+   int idhi[3] = {AMREX_D_DECL(hi.x,hi.y,hi.z)};
+   int idlo_e[3] = {AMREX_D_DECL(lo.x,lo.y,lo.z)};          // extended box bounds
+   int idhi_e[3] = {AMREX_D_DECL(hi.x,hi.y,hi.z)};
+   for (int dim = 0; dim < AMREX_SPACEDIM; dim++) {   // Extend only AMREX_SPACEDIMs
+      idlo_e[dim] -= 1;
+      idhi_e[dim] += 1;
+   }
+
+   int k = idlo[2];
+
+   for (int n = 0; n < nComp; n++) {
+
+      // set all crse cells to zero first
+      for (int j = idlo_e[1]; j <= idhi_e[1]; ++j) {
+         for (int i = idlo_e[0]; i <= idhi_e[0]; ++i) {
+            if (mask(i,j,k) == crsecell) data(i,j,k,n) = 0.0_rt;
```

Comment:
In the GPU case, `bx` has only a single cell and a thread works on a single cell.  This line will cause race conditions.

Reply:
Can I just use two kernels ?

```
amrex::ParallelFor(amrex::grow(bx,1), ncomp,
[=] AMREX_GPU_DEVICE (int i, int j, int k, int n) noexcept
{   
   if (mask_arr(i,j,k) == crsebnd) data_arr(i,j,k,n) = 0.0;
}); 

AMREX_LAUNCH_HOST_DEVICE_LAMBDA(bx, tbx,
{   
    amrex_first_order_extrap(tbx, ncomp, mask_arr, data_arr);
}); 
```
and not have the first nested loops in the `amrex_first_order_extrap` or we actually have the problem throughout the entire function ? Note that I might just be lucky, but I did not ran into troubles running the code in its current version ... (on both Garuda and Ascent)
Since we actually only care about the ghost cells here, maybe I should go with something like what you did `GpuBndryFuncFab<F>::ccdoit` ?